File: Palindrome Linked List.js
var isPalindrome = function(head) {
    var arr = [];
    var current = head;
    while(current){
        arr.push(current.val);
        current = current.next;
    }
    var start = 0;
    var end = arr.length-1;
    while(start<end){
        if(arr[start] !== arr[end]) return false;
        start++;
        end--;
    }
    return true;
};

File: Palindrome Linked List.py
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        values = []
        while head:
            values.append(head.val)
            head = head.next
        l, r = 0, len(values) -1
        while l < r:
            if values[l] != values[r]:
                return False
            l += 1
            r -= 1
        return True

            
